src/wazuh_modules/vulnerability_scanner/CMakeLists.txt

```diff
@@ -104,7 +104,13 @@ add_library(vulnerability_scanner SHARED
     )
 target_compile_options(vulnerability_scanner PRIVATE -Wall -Wextra -Wshadow -Wnon-virtual-dtor -Woverloaded-virtual -Wunused -Wcast-align -Wformat=2)
 
-add_dependencies(database_feed compile_schemas_vd compile_schemas_input) #Add a dependency between top-level targets.
+if (TARGET compile_schemas_input)
+    add_dependencies(database_feed compile_schemas_vd compile_schemas_input)
+else()
+    # If target isn't defined, only this project is being built and not the whole Wazuh project.
+    # The schemas must be compiled manually before building this project in that case.
+    add_dependencies(database_feed compile_schemas_vd)
+endif()
```

Comment:
This assume you're compiling the schemas manually outside this project (as clearly mentioned in the comment) but using this, isn't the `compile_schemas_input` target always missing?

Reply:
The only case when the target is present is when the whole project is being built (`make TARGET=server`) because the general CMakeLists.txt file has all the `add_subdirectory` calls required to "discover" the target.

Reply:
Yeap, exactly. Thanks